vectorstores/pgvector/engine.go

```diff
@@ -0,0 +1,141 @@
+package pgvector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/oauth2/v2"
+	"google.golang.org/api/option"
+)
+
+const Port = 5432 // default PostgreSQL port
+
+type ServiceAccountRetriever interface {
+	GetServiceAccountEmail(ctx context.Context) (string, error)
+}
+type DefaultServiceAccountRetriever struct{}
+
+func (d *DefaultServiceAccountRetriever) GetServiceAccountEmail(ctx context.Context) (string, error) {
+	return getServiceAccountEmail(ctx)
+}
+
+type Column struct {
+	Name     string
+	DataType string
+	Nullable bool
+}
+
+func NewColumn(name, dataType string, nullable bool) (*Column, error) {
+	if name == "" {
+		return nil, errors.New("name should be provided")
+	}
+	if dataType == "" {
+		return nil, errors.New("dataType should be provided")
+	}
+	return &Column{Name: name, DataType: dataType, Nullable: nullable}, nil
+}
+
+type PostgresEngine struct {
+	Host                    string
+	Port                    int
+	Database                string
+	User                    string
+	Password                string
+	IAMAccountEmail         string
+	ServiceAccountRetriever ServiceAccountRetriever
+}
+
+// NewPostgresEngine initializes a new PostgresEngine
+func NewPostgresEngine(host, dbname, user, password, iamAccountEmail string, port int) *PostgresEngine {
+	return &PostgresEngine{
+		Host:            host,
+		Port:            port,
+		Database:        dbname,
+		User:            user,
+		Password:        password,
+		IAMAccountEmail: iamAccountEmail,
+	}
+}
+
+// CreatePostgresEngine establishes a connection to the PostgreSQL database.
+func (p *PostgresEngine) CreatePostgresEngine(ctx context.Context) (*pgx.Conn, error) {
+	err := p.assignUser(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error assigning user. Err: %s", err)
+	}
+
+	// Build the connection configuration
+	config, _ := pgx.ParseConfig("")
+	config.Host = p.Host
+	config.Port = uint16(p.Port)
+	config.Database = p.Database
+	config.User = p.User
+	config.Password = p.Password
+
+	// Make the connection to the PostgreSQL database
+	connection, err := pgx.ConnectConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to Database: %s, User: %s, Port: %v, Err: %s", p.Database, p.User, p.Port, err)
```

Comment:
Should wrap the error (https://go.dev/blog/go1.13-errors#wrapping-errors-with-w) here instead of the string value (I think the other information is probably too verbose)
```suggestion
		return nil, fmt.Errorf("unable to connect: %w", err)
```